---
title: "R_script"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

#Import transcript-level abundance and generate the matrix of counts at gene level
```{r tximport eval=F echo=T}
#Install biomart if necessary
#if (!requireNamespace("BiocManager", quietly = TRUE))
#   install.packages("BiocManager")
#BiocManager::install("biomaRt")

#Loading the librairies of interest
library("tximport")
library("biomaRt")

#Working in the right directory
setwd("/ifb/data/mydatalocal/tp_ngs_single_cell")

#Taking the path of the directory with salmon data (i.e. "alignments")
dir="/ifb/data/mydatalocal/data/alignments"

#Create a list of files in the directory with salmon data (i.e. "alignments")
files=list.files(dir)

#Take the names of the cells from the directory with salmon data (i.e. the names of cells in "alignments")
cells <- gsub(".fastq.gz_quant","",files)

#Create a list with the paths toward the quant.sf files in the directory with salmon data. These paths' names are the names of cells.
files=paste0(dir,"/",files,"/","quant.sf")
names(files)=cells
files[1:10] #Display the 10 first paths

#Annotating the transcripts with the names of genes that we want
ensembl <- useEnsembl(biomart = "genes", dataset = "mmusculus_gene_ensembl")          #select the genes of mus musculus in Ensembl
attributeNames <- c('ensembl_gene_id', 'external_gene_name','ensembl_transcript_id')   #create the vector that will contain the 
annot <- getBM(attributes=attributeNames, 
               mart = ensembl)            #extract and make the correspondence between the external gene name and the transcript id
tx2gene <- data.frame(txname=annot$"ensembl_transcript_id",geneid=annot$"external_gene_name")  #create a frame with the corresponding name of the gene for every transcript. It will be the argument for tximport.
head(tx2gene) 

#Import and summarize the transcript-level abundance from the salmon data (splitted to improve the computer capacity)
txi1 <- tximport(files[1:500],type="salmon",tx2gene=tx2gene,ignoreTxVersion=T)
txi2 <- tximport(files[501:1000],type="salmon",tx2gene=tx2gene,ignoreTxVersion=T)
txi3 <- tximport(files[1001:1500],type="salmon",tx2gene=tx2gene,ignoreTxVersion=T)
txi4 <- tximport(files[1501:2000],type="salmon",tx2gene=tx2gene,ignoreTxVersion=T)
txi5 <- tximport(files[2001:2553],type="salmon",tx2gene=tx2gene,ignoreTxVersion=T)

#Pull the splitted results in one single variable
txi <- txi1
txi$"abundance" <- cbind(txi1$"abundance",txi2$"abundance",txi3$"abundance",txi4$"abundance",txi5$"abundance")
txi$"counts" <- cbind(txi1$"counts",txi2$"counts",txi3$"counts",txi4$"counts",txi5$"counts")
txi$"length" <- cbind(txi1$"length",txi2$"length",txi3$"length",txi4$"length",txi5$"length")
txi$"countsFromAbundance" <- cbind(txi1$"countsFromAbundance",txi2$"countsFromAbundance",txi3$"countsFromAbundance",txi4$"countsFromAbundance",txi5$"countsFromAbundance")

#Save the content in a .rds file in the current directory
saveRDS(txi, "txi.rds")
```


#Install required librairies and create Seurat object
```{r create_seurat, echo=T}
library("Seurat")
setwd("/ifb/data/mydatalocal/tp_ngs_single_cell")
#Load the matrix of counts
txidata <-readRDS("txi.rds")
#Create the seurat object that will contain the data and the analysis
incisor <- CreateSeuratObject(counts = txidata$counts, project = "incisor", min.cells = 3, min.features = 200)
```

#Violin plot of cells' features (raw data)
```{r Vilolin_plots, echo=T}
#add a column to the matrix that will include the % of mitDNA
incisor[["percent.mt"]] <- PercentageFeatureSet(incisor, pattern = "^mt-")
#plot the number of RNA counts, the percentage of mitochondrial gene present in cells and the number of unique genes for each cell
plot_RNA <-VlnPlot(incisor, features = c("percent.mt", "nFeature_RNA", "nCount_RNA"), ncol = 3)
```

#Plot of one feature against another
```{r Feature_plots, echo=T}
plot1 <- FeatureScatter(incisor, feature1 = "nCount_RNA", feature2 = "percent.mt")
plot_ncount_against_nfeatures <- FeatureScatter(incisor, feature1 = "nCount_RNA", feature2 = "nFeature_RNA")
plot1 + plot_ncount_against_nfeatures
```

#Subselection of the data
# We select cells with: <15% mitochondrial genes, <1 million RNAs and between the 5th and 95th centile of unique genes in cells  
```{r subset, echo=T}
#Calculate the quantile
quantile5_incisor <- quantile(incisor$nFeature_RNA, probs = c(0.05), na.rm = FALSE, names = TRUE, type = 7)
#Make the sub-selection
incisor <- subset(incisor, subset = nFeature_RNA > quantile5_incisor & percent.mt < 15 & nCount_RNA < 1000000)
```

#Normalization
#We normalize the gene expression measurements for each cell by the total expression
```{r normalization, echo=T}
incisor <-NormalizeData(incisor, normalization.method = "LogNormalize", scale.factor = 10000)
```

#Identification of the variable genes between cells
```{r identify_variable_features echo=T}
#Take the 2000 more variable genes
incisor <-FindVariableFeatures(incisor, selection.method = "vst", nfeatures = 2000)
#Store the 10th first more variable genes
top10 <- head(VariableFeatures(incisor), 10)
```

#Plot the variable genes
```{r plot_variable_genes echo=T}
plot1 <- VariableFeaturePlot(incisor)
plot2 <- LabelPoints(plot = plot1, points = top10, repel = TRUE) #Evidence the names of the 10 most variable genes
plot2
```

#Scaling the data
```{r scaling echo=T}
all.genes <- rownames(incisor)
incisor <- ScaleData(incisor, features = all.genes)
```

#Dimensional reduction
```{r dimensional_reduction echo=T}
#Run PCA
incisor <- RunPCA(incisor, features = VariableFeatures(object = incisor))
#Examine and visualize PCA results
print(incisor[["pca"]], dims = 1:5, nfeatures = 5) #Give 5 genes that explain the most the variability through 5 dimensions
VizDimLoadings(incisor, dims = 1:2, reduction = "pca")
DimPlot(incisor, reduction = "pca")
DimHeatmap(incisor, dims = 1, cells = 500, balanced = TRUE)
DimHeatmap(incisor, dims = 1:15, cells = 500, balanced = TRUE)
```

#Determine the dimensionality of the dataset
```{r dimensionality echo=T}
incisor <- JackStraw(incisor, num.replicate = 100)
incisor <- ScoreJackStraw(incisor, dims = 1:20)
JackStrawPlot(incisor, dims = 1:20)
ElbowPlot(incisor) #Gives the variability explained according to the number of dimensions taken
```

#Cluster the cells
```{r cluster echo=T}
incisor <- FindNeighbors(incisor, dims = 1:20)
incisor <- FindClusters(incisor, resolution = 0.5)
# Look at cluster IDs of the first 5 cells
head(Idents(incisor), 5)
```

#Run the UMAP
```{r umap echo=T}
incisor <-RunUMAP(incisor, dims=1:10)
DimPlot(incisor, reduction = "umap")
saveRDS(incisor, "cluster.rds")

d=DimPlot(incisor, reduction = "umap")
d
ggsave(file="umap.png",d)

#Find the markers for clusters
  #find all markers of cluster 1
cluster1.markers <- FindMarkers(incisor, ident.1 = 1, min.pct = 0.25)
head(cluster1.markers, n = 5)
  #find all markers distinguishing cluster 5 from clusters 0 and 3
cluster5.markers <- FindMarkers(incisor, ident.1 = 5, ident.2 = c(0, 3), min.pct = 0.25)
head(cluster5.markers, n = 5)
  #find markers for every cluster compared to all remaining cells, report only the positive ones
incisor.markers <- FindAllMarkers(incisor, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
incisor.markers %>% group_by(cluster) %>% top_n(n = 2, wt = avg_log2FC)
  #Test Rock 
cluster1.markers <- FindMarkers(incisor, ident.1 = 0, log2FC.threshold = 0.25, test.use = "roc", only.pos = TRUE)
  #Display the violin plot for the given genes 
VlnPlot(incisor, features = c("MS4A1", "CD79A"))
  #Display the violin plot of raw counts for the given genes
VlnPlot(incisor, features = c("NKG7", "PF4"), slot = "counts", log = TRUE)
