---
title: "R_script"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## R Markdown

This is an R Markdown document. Markdown is a simple formatting syntax for authoring HTML, PDF, and MS Word documents. For more details on using R Markdown see <http://rmarkdown.rstudio.com>.

When you click the **Knit** button a document will be generated that includes both content as well as the output of any embedded R code chunks within the document. You can embed an R code chunk like this:

```{r tximport eval=F echo=T}
#if (!requireNamespace("BiocManager", quietly = TRUE))
#   install.packages("BiocManager")
#BiocManager::install("biomaRt")
library("tximport")
library("biomaRt")

setwd("/ifb/data/mydatalocal/tp_ngs_single_cell")
dir="/ifb/data/mydatalocal/data/alignments"
files=list.files(dir)
cells <- gsub(".fastq.gz_quant","",files)
files=paste0(dir,"/",files,"/","quant.sf")
names(files)=cells
files[1:10]
ensembl <- useEnsembl(biomart = "genes", dataset = "mmusculus_gene_ensembl")
attributeNames <- c('ensembl_gene_id', 'external_gene_name','ensembl_transcript_id')
annot <- getBM(attributes=attributeNames, 
               mart = ensembl)
tx2gene <- data.frame(txname=annot$"ensembl_transcript_id",geneid=annot$"external_gene_name")
head(tx2gene)
txi1 <- tximport(files[1:500],type="salmon",tx2gene=tx2gene,ignoreTxVersion=T)
txi2 <- tximport(files[501:1000],type="salmon",tx2gene=tx2gene,ignoreTxVersion=T)
txi3 <- tximport(files[1001:1500],type="salmon",tx2gene=tx2gene,ignoreTxVersion=T)
txi4 <- tximport(files[1501:2000],type="salmon",tx2gene=tx2gene,ignoreTxVersion=T)
txi5 <- tximport(files[2001:2553],type="salmon",tx2gene=tx2gene,ignoreTxVersion=T)

txi <- txi1
txi$"abundance" <- cbind(txi1$"abundance",txi2$"abundance",txi3$"abundance",txi4$"abundance",txi5$"abundance")
txi$"counts" <- cbind(txi1$"counts",txi2$"counts",txi3$"counts",txi4$"counts",txi5$"counts")
txi$"length" <- cbind(txi1$"length",txi2$"length",txi3$"length",txi4$"length",txi5$"length")
txi$"countsFromAbundance" <- cbind(txi1$"countsFromAbundance",txi2$"countsFromAbundance",txi3$"countsFromAbundance",txi4$"countsFromAbundance",txi5$"countsFromAbundance")

saveRDS(txi, "txi.rds")
```

## Including Plots

You can also embed plots, for example:

```{r pressure, echo=FALSE}
plot(pressure)
```

Note that the `echo = FALSE` parameter was added to the code chunk to prevent printing of the R code that generated the plot.
